src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/_params.py

```diff
@@ -3,43 +3,54 @@
     with self.argument_context('redis') as c:
-        c.argument('name', options_list=['--name', '-n'], help='Name of the Redis cache.', id_part='name',
-                   completer=get_resource_name_completion_list('Microsoft.Cache/redis'))
+        cache_name = CLIArgumentType(options_list=['--name', '-n'], help='Name of the Redis cache.', id_part='name',
+                                     completer=get_resource_name_completion_list('Microsoft.Cache/redis'))
+        # pylint:disable=line-too-long
+        cache_name_without_id_part = CLIArgumentType(options_list=['--name', '-n'], help='Name of the Redis cache.', id_part=None,
+                                                     completer=get_resource_name_completion_list('Microsoft.Cache/redis'))
+        format_type = CLIArgumentType(options_list=['--file-format'], help='Format of the blob (Currently rdb is the only supported format, with other formats expected in the future)')
+
+        c.argument('name', arg_type=cache_name)
         c.argument('redis_configuration', help='JSON encoded configuration settings. Use @{file} to load from a file.',
                    type=JsonString)
         c.argument('reboot_type', arg_type=get_enum_type(RebootType))
         c.argument('key_type', arg_type=get_enum_type(RedisKeyType))
         c.argument('shard_id', type=int)
         c.argument('sku', help='Type of Redis cache.', arg_type=get_enum_type(SkuName))
         c.argument('vm_size', help='Size of Redis cache to deploy. Example : values for C family (C0, C1, C2, C3, C4, '
-                                   'C5, C6). For P family (P1, P2, P3, P4)')
+                                   'C5, C6). For P family (P1, P2, P3, P4, P5)')
         c.argument('enable_non_ssl_port', action='store_true')
         c.argument('shard_count', type=int)
-        c.argument('tags', tags_type)
-
-    for scope in ['redis import-method', 'redis import']:
-        with self.argument_context(scope) as c:
-            c.argument('files', nargs='+', help='Space-separated list of files to import.')
-            c.argument('file_format', help='File format to import.')
-
-    with self.argument_context('redis patch-schedule set') as c:
-        c.argument('schedule_entries', type=ScheduleEntryList)
-
-    with self.argument_context('redis create') as c:
-        c.argument('name', arg_type=name_type, completer=None)
+        c.argument('files', help='SAS url for blobs that needs to be imported', nargs='+')
+        c.argument('format', arg_type=format_type)
+        c.argument('file_format', arg_type=format_type)
+        c.argument('container', help='SAS url for container where data needs to be exported to')
+        c.argument('prefix', help='Prefix to use for exported files')
+        # pylint: disable=line-too-long
+        c.argument('schedule_entries', help="List of Patch schedule entries. Example Value:[{\"dayOfWeek\":\"Monday\",\"startHourUtc\":\"00\",\"maintenanceWindow\":\"PT5H\"}]", type=ScheduleEntryList)
```

Comment:
It would be good to show one or two examples of how to use this command. In _help.py.  
One of the examples could be like you have shown  here.
Another could mention that it is possible to store the json input in a file and us --schedule-entries @entries.json
See role/_help.py for an example showing how `@` is used. Search for @ad-role.json.

cc/ @tjprescott we did not come to a conclusion on how we would be dealing with this parameter.